flink-staging/flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala

```diff
@@ -61,99 +73,217 @@ object FlinkShell {
         )
 
       cmd("remote") action { (_, c) =>
-        c.copy(flinkShellExecutionMode = ExecutionMode.REMOTE)
+        c.copy(executionMode = ExecutionMode.REMOTE)
       } text("starts Flink scala shell connecting to a remote cluster\n") children(
         arg[String]("<host>") action { (h, c) =>
-          c.copy(host = h) }
+          c.copy(host = Some(h)) }
           text("remote host name as string"),
         arg[Int]("<port>") action { (p, c) =>
-          c.copy(port = p) }
+          c.copy(port = Some(p)) }
           text("remote port as integer\n"),
         opt[(String)]("addclasspath") abbr("a") valueName("<path/to/jar>") action {
           case (x, c) =>
             val xArray = x.split(":")
             c.copy(externalJars = Option(xArray))
-          } text("specifies additional jars to be used in Flink")
+        } text ("specifies additional jars to be used in Flink")
       )
-      help("help") abbr("h") text("prints this usage text\n")
+
+      cmd("yarn") action {
+        (_, c) => c.copy(executionMode = ExecutionMode.YARN, yarnConfig = None)
+      } text ("starts Flink scala shell connecting to a yarn cluster\n") children(
+        opt[Int]("container") abbr ("n") valueName ("arg") action {
+          (x, c) =>
+            c.copy(yarnConfig = Some(ensureYarnConfig(c).copy(containers = Some(x))))
+        } text ("Number of YARN container to allocate (= Number of Task Managers)"),
+        opt[Int]("jobManagerMemory") abbr ("jm") valueName ("arg") action {
+          (x, c) =>
+            c.copy(yarnConfig = Some(ensureYarnConfig(c).copy(jobManagerMemory = Some(x))))
+        } text ("Memory for JobManager Container [in MB]"),
+        opt[String]("name") abbr ("nm") action {
+          (x, c) => c.copy(yarnConfig = Some(ensureYarnConfig(c).copy(name = Some(x))))
+        } text ("Set a custom name for the application on YARN"),
+        opt[String]("queue") abbr ("qu") valueName ("<arg>") action {
+          (x, c) => c.copy(yarnConfig = Some(ensureYarnConfig(c).copy(queue = Some(x))))
+        } text ("Specify YARN queue"),
+        opt[Int]("slots") abbr ("s") valueName ("<arg>") action {
+          (x, c) => c.copy(yarnConfig = Some(ensureYarnConfig(c).copy(slots = Some(x))))
+        } text ("Number of slots per TaskManager"),
+        opt[Int]("taskManagerMemory") abbr ("tm") valueName ("<arg>") action {
+          (x, c) =>
+            c.copy(yarnConfig = Some(ensureYarnConfig(c).copy(taskManagerMemory = Some(x))))
+        } text ("Memory per TaskManager Container [in MB]"),
+        opt[(String)] ("addclasspath") abbr("a") valueName("<path/to/jar>") action {
+          case (x, c) =>
+            val xArray = x.split(":")
+            c.copy(externalJars = Option(xArray))
+        } text("specifies additional jars to be used in Flink\n")
+      )
+
+      help("help") abbr ("h") text ("prints this usage text\n")
```

Comment:
All other descriptions start with a capital letter. Should be uniformly handled.